### PR TITLE
feat: support glyphs

### LINKED ISSUE #5: Add routes, api methods & e2e tests for Glyphs
Mapbox GL maps require fonts rendered to SDFs (Signed-distance field), this makes rendering in GL fast and pretty. We need to [support this route](https://docs.mapbox.com/api/maps/fonts/):

- [ ] `GET /fonts/{font}/{start}-{end}.pbf` - where `{font}` can be a comma-separated (URL encoded) list of font names, and we should return the first available (similar to how a CSS `font-family` property works).

It is possible to render SDFs dynamically (e.g. https://github.com/mapbox/node-fontnik) but I'm wary about introducing another native library in our stack. We can download them from the Mapbox API, but that's not a great solution either.

I suggest that we generate SDFs for common fonts as part of our build process and ship them as static files included in the npm bundle. We can use node-fontnik (or maybe there is a better maintained SDF renderer out there?) for generating these at build time. We would probably need to stick to open fonts that we are able to bundle in the app. We should support some kind of fallback lookup table, so that if we do not have a font that is referenced in the `{font}` part of the URL, then we fall back to the closest equivalent.

We should also support downloading SDFs / glyphs from the Mapbox API when the user downloads a style that references a font that we do not have pre-rendered.

Maybe as an MVP, we proxy / download Mapbox rendered SDFs directly from their API. We should store these in a way that they can be shared across different styles, e.g. if we use the SWR cache then we should normalize the URL to reduce the `{font}` part to a single font name.

### PR DESCRIPTION
**Pull request by gmaclennan**
Closes #5 

@achou11 here's a quick script to generate SDFs - just wanted to check it worked and start my Monday with something with a quick result :)

### CODE REVIEW

File: scripts/generate-sdf.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const fs = require('fs')
```
Comment (achou11) [inline scripts/generate-sdf.js:1]: nit, but add `#!/usr/bin/env node` as first line?

File: scripts/generate-sdf.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+const outDir = path.join(__dirname, '../sdf/opensans')
+mkdirp.sync(outDir)
+
+generateSdf({ font, outDir })
+
+async function generateSdf({ font, outDir }) {
+  return new Promise((resolve, reject) => {
+    for (let i = 0; i < 65536; i = i + 256) {
```
Comment (achou11) [inline scripts/generate-sdf.js:17]: for my own education, what are these numbers?

**Review by achou11 — APPROVED** [review]
if it works, I won't dispute that 😄 had a couple of small comments but otherwise seems safe to merge and use in a follow up

File: scripts/generate-sdf.js (hunk 1 above)
Comment (gmaclennan) [reply scripts/generate-sdf.js:1]: Happy for you to add that. Would also need to make it executable. I tend towards keeping scripts without a hashbang when they are not being published as scripts to run outside the module - just to always run them with `node ./script.js`

Comment (achou11) [reply scripts/generate-sdf.js:1]: good point! not necessary in this case

File: scripts/generate-sdf.js (hunk 2 above)
Comment (gmaclennan) [reply scripts/generate-sdf.js:17]: These number are character codes. The [mapbox api](https://docs.mapbox.com/api/maps/fonts/) expects SDFs to be served as combined files of 256 glyphs (see the API details). For ASCII the map only needs to request the 0-255.pbf, but when the client library (mapbox-gl-js) comes across labels with other characters, it downloads the glyph ranges needed to render those characters.

**Comment by achou11** [comment]
@gmaclennan marked as ready for review. probably missing a few key tests and some refinements, but think it's ready to be glanced at

File: src/api/glyphs.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        res({ type: 'file', data: staticPath })
+      })
+    })
+  }
+
+  return {
+    // TODO: Should we return always return the offline asset if it exists?
+    async getGlyphs({ styleId, accessToken, font, start, end }) {
```
[the remark this answers is not in the document]
Comment (achou11) [reply src/api/glyphs.ts:54]: see https://github.com/digidem/mapeo-map-server/pull/54/commits/13f0bf353f233108415198aea5be5766fc8d1f5e for offline handling. in this case we serve the default static asset

**Pull request merged**